File: source/com/barrybecker4/common/expression/Tokens.java
// Copyright by Barry G. Becker, 2013. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.common.expression;

/**
 * @author Barry Becker
 */
public enum Tokens {

    PLUS('+'),
    MINUS('-'),
    LEFT_PAREN('('),
    RIGHT_PAREN(')');

    private char symbol;

    Tokens(char c) {
       symbol = c;
    }

    public char getSymbol() {
        return symbol;
    }

}

File: scala-source/com/barrybecker4/common/expression/Tokens.scala
/* Copyright by Barry G. Becker, 2000-2018. Licensed under MIT License: http://www.opensource.org/licenses/MIT */
package com.barrybecker4.common.expression

abstract sealed class Token(val symbol: Char)

case object PLUS extends Token('+')
case object MINUS extends Token('-')
case object LEFT_PAREN extends Token('(')
case object RIGHT_PAREN extends Token('+')
